checkov/common/util/secrets.py

```diff
@@ -139,21 +140,29 @@ def omit_secret_value_from_line(secret: str | None, line_text: str) -> str:
     return censored_line
 
 
-def omit_secret_value_from_checks(check: BaseCheck, check_result: dict[str, CheckResult] | _CheckResult,
-                                  entity_code_lines: list[tuple[int, str]],
-                                  entity_config: dict[str, Any] | ParameterAttributes | ResourceAttributes,
-                                  resource_attributes_to_omit: ResourceAttributesToOmit | None = None) -> \
-        list[tuple[int, str]]:
-    secrets = set()  # a set, to efficiently avoid duplicates in case the same secret is found in the following conditions
+def omit_secret_value_from_checks(
+        check: BaseCheck,
+        check_result: dict[str, CheckResult] | _CheckResult,
+        entity_code_lines: list[tuple[int, str]],
+        entity_config: dict[str, Any] | ParameterAttributes | ResourceAttributes,
+        resource_attributes_to_omit: ResourceAttributesToOmit | None = None
+) -> list[tuple[int, str]]:
+    # a set, to efficiently avoid duplicates in case the same secret is found in the following conditions
+    secrets = set()
     censored_code_lines = []
 
     if CheckCategories.SECRETS in check.categories and check_result.get('result') == CheckResult.FAILED:
         secrets.update([str(secret) for key, secret in entity_config.items() if
                         key.startswith(f'{check.id}_secret')])
 
-    if resource_attributes_to_omit and check.entity_type in resource_attributes_to_omit:
-        for attribute_to_omit in [attr for attr in resource_attributes_to_omit.get(check.entity_type) if attr in entity_config]:  # type:ignore[union-attr]
-            secret = entity_config.get(attribute_to_omit)
+    if resource_attributes_to_omit:
+        universal_mask = resource_attributes_to_omit.get(RESOURCE_ATTRIBUTES_TO_OMIT_UNIVERSAL_MASK, set())
+        resource_masks = resource_attributes_to_omit.get(check.entity_type, set())
+        resource_masks.update(universal_mask)
```

Comment:
```suggestion
        resource_masks = resource_attributes_to_omit.get(check.entity_type, set())
        if RESOURCE_ATTRIBUTES_TO_OMIT_UNIVERSAL_MASK in resource_attributes_to_omit:
            resource_masks.update(resource_attributes_to_omit[RESOURCE_ATTRIBUTES_TO_OMIT_UNIVERSAL_MASK])
```

As most of the runs won't have the RESOURCE_ATTRIBUTES_TO_OMIT_UNIVERSAL_MASK (it's a specific opt in case by users), I think its better to avoid the set update + variable saving when its not relevant

Reply:
I don't totally under stand :| 
I get both universal mask & resource mask - resource mask with universal mask (set update)
in case universal mask is not in the resource_attributes_to_omit it will update set with the empty set (o(1))
@arielkru

Reply:
@pazbechor I agree it's o(1) so no big problem here, just thought of a tiny performance improvement so it works either way.
Let's say 90% of the runs won't have a global attribute, so as the code is now, these 3 instructions will always happen:
1. get `'*'` from`resource_attributes_to_omit`
2. save an empty set to `universal_mask` variable
3. update `resource_masks` with an empty set

So for 90% of the cases (just a rough assumption) the last 2 instructions are redundant and we can skip them by just checking first #1 and if its None, don't do the other 2.
But again, just a minor improvement

Reply:
we talked about it :)